### ISSUE TITLE
php 8.1 deprecation

### ISSUE THREAD

**Issue by jeckel** [gggeek/phpxmlrpc#89]
Running with **php8.1 rc 5** throw some deprecation warning :

```
<br />
<b>Deprecated</b>: Return type of PhpXmlRpc\Value::count() should either be compatible with Countable::count(): int, or
the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in
<b>/application/vendor/phpxmlrpc/phpxmlrpc/src/Value.php</b> on line <b>492</b><br />
<br />
<b>Deprecated</b>: Return type of PhpXmlRpc\Value::getIterator() should either be compatible with
IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily
suppress the notice in <b>/application/vendor/phpxmlrpc/phpxmlrpc/src/Value.php</b> on line <b>512</b><br />
<br />
<b>Deprecated</b>: Return type of PhpXmlRpc\Value::offsetExists($offset) should either be compatible with
ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily
suppress the notice in <b>/application/vendor/phpxmlrpc/phpxmlrpc/src/Value.php</b> on line <b>577</b><br />
<br />
<b>Deprecated</b>: Return type of PhpXmlRpc\Value::offsetGet($offset) should either be compatible with
ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily
suppress the notice in <b>/application/vendor/phpxmlrpc/phpxmlrpc/src/Value.php</b> on line <b>620</b><br />
<br />
<b>Deprecated</b>: Return type of PhpXmlRpc\Value::offsetSet($offset, $value) should either be compatible with
ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to
temporarily suppress the notice in <b>/application/vendor/phpxmlrpc/phpxmlrpc/src/Value.php</b> on line <b>532</b><br />
<br />
<b>Deprecated</b>: Return type of PhpXmlRpc\Value::offsetUnset($offset) should either be compatible with
ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily
suppress the notice in <b>/application/vendor/phpxmlrpc/phpxmlrpc/src/Value.php</b> on line <b>597</b><br />
```

`Countable` and `ArrayAccess` now needs to have defined return types.

Fix can be done by adding the return types to each method signature (but can breaks with older version of php, like `mixed` return is only avilable since php7.4) or by adding the attribute `#[\ReturnTypeWillChange]` (which is ugly, but doesn't break with older version of php.

**Comment by gggeek** [comment]
Thx for the report.
I think we will go for the ugly attribute hack. BC is priority one for this lib...

**Comment by gggeek** [comment]
Fixed in master branch.

I am currently moving Ci from Travis to GitHub Actions, will release as soon as that is ok.

**Comment by gggeek** [comment]
Released in 4.6.0
